### PR TITLE
done with DL training & inference

### PR DESCRIPTION
**Pull request by choonghwanlee**
1. Siamese Network with CLIP backbone, trained using contrastive loss on only the final transformer layer of the CLIP model.
2. a custom projection layer is added to reduce the dimensionality of the embeddings to size 128
3. trained over 3 epochs with learning rate scheduler, achieved validation accuracy (roc_auc) of 0.89
4. we pre-store the embeddings of our database into a FAISS index for O(N) queries at inference time vs. O(N^2)

you can find the inference script at `inference_siamese.py`

### CODE REVIEW

**Comment by ludaladila** [comment]
Looks great! I really like the use of contrastive loss with a CLIP backbone and FAISS for efficient retrieval. I’ll build the app on the inference—thanks for the solid foundation!

**Comment by KelllyChen** [comment]
The setup looks solid, and it's pretty good to get 0.89 on roc_auc in three epochs
